### DISCUSSION TITLE
Automation only triggered once every few attempts

### DISCUSSION THREAD

**Discussion by vaclaff** [Budibase/budibase#7029] [Help] [answered]
**Hosting**
- Cloud, latest version

**Describe the bug**
I have automation that is triggered by an app action (i.e. button click in a table) with an input (row's _id) and updates the relevant row subsequently (only changes one boolean). It works flawlessly when testing from the builder, but when running from the app frontend (i.e. actually clicking the button), the automation is only performed once in a while.

It may sound weird, but I try to run the automation like 20 times, but in the automation history, I only see two attempts, even though I saw a green notification for every single attempt in the frontend.

I assume there is neither a problem with the automation itself nor with the transfer of relevant row's _id as a parameter. Everything works fine during testing and even in the published app if one gets lucky and the automation is actually executed.

**Expected behavior**
The automation should run every time I click the button, not only once in a while.

**Comment by vaclaff** [comment]
I refreshed the Automation history after posting this issue and I realized that a bunch of attempts were executed in one moment.

Could the issue be connected to the fact that I'm using cloud version with a free plan (planning to move to a self hosted version, but the cloud is very convenient for development)...

**Comment by mike12345567** [comment, marked as answer]
Hi @vaclaff - yes there are a few things at play here, your automations will be queued up in the Cloud alongside every other automation that a user has requested to be run - there is no guarantee of when it will infact be executed; it should be relatively soon after you've requested it but it may not be instant.

Self hosted should be faster in this way but the automation system is designed currently to be fully asynchronous, you can trigger an automation but it will be added to a queue for execution and if the queue has a lot on it already (say 20 automations triggered at once) they will all be carried out individually, one by one.

I've moved this from an issue to a discussion as it does not appear to be an issue with the platform.

**Reply by vaclaff** [reply]
Thank you for a very quick reaction! Now I understand... It initially seemed to me like a bug, glad it's already moved.
